control/init.sh

```diff
@@ -10,20 +11,29 @@ then
 fi
 
 
-function realpath() {
-    python3 -c "import os,sys; print(os.path.realpath(sys.argv[1]))" $1
-}
-
-
 if [ -z ${CI_TEST} ]; then
-    if [ ! -f $VENV/bin/activate ]; then
-        # use virtualenv because `python3 -m venv` is broken on Ubuntu 18.04
-        python3 -m pip install --user --upgrade virtualenv
-        python3 -m virtualenv $VENV
+    if [ "$CCHQ_PYTHON" == 310 ]; then
+      if [ ! -f $VENV/bin/activate ]; then
+          # use virtualenv because `python3 -m venv` is broken on Ubuntu 18.04
+          python3.10 -m pip install --user --upgrade virtualenv
+          python3.10 -m virtualenv $VENV
+      fi
+    else
+      if [ ! -f $VENV/bin/activate ]; then
+          # use virtualenv because `python3 -m venv` is broken on Ubuntu 18.04
+          python3 -m pip install --user --upgrade virtualenv
+          python3 -m virtualenv $VENV
+      fi
     fi
     source $VENV/bin/activate
 fi
 
+
+function realpath() {
+    python -c "import os,sys; print(os.path.realpath(sys.argv[1]))" $1
+}
```

Comment:
might have been unnecessary to move this method until after the virtualenv was activated, but wanted to ensure `python` was referencing the correct binary.

Reply:
This should work with any version of Python.